dame_flame/flame_algorithm.py

```diff
@@ -127,6 +127,12 @@ def flame_generic(df_all, treatment_column_name, weight_array,
     all_covs.remove(outcome_column_name)
     df_unmatched = df_all.copy(deep=True) 
     
+    #Ensuring that no entries are non-integers
+    for index, row in df_all.iterrows():
+        for i in all_covs:
+            assert (type(row[i])==int), 'Input has non-integer entries'
+            
```

Comment:
Same comment as in the dame_algorithm.py file. If it's done in data_cleaning then it's in an appropriately named spot and also minimizes repetitive code between these two files. (That file is more than just cleaning, its general pre-processing and should maybe be renamed accordingly....there should be functions there that check other stuff like non-binary treatment and this check is somewhat in the same category)